debug/handler/interceptorTxDebug.go

```diff
@@ -0,0 +1,91 @@
+package handler
+
+import (
+	"encoding/hex"
+	"sync"
+	"time"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+
+	p2p2 "github.com/multiversx/mx-chain-communication-go/p2p"
+	"github.com/multiversx/mx-chain-go/p2p"
+	"github.com/multiversx/mx-chain-go/process"
+)
+
+const (
+	interceptedTx         = "intercepted tx"
+	interceptedRewardTx   = "intercepted reward tx"
+	interceptedUnsignedTx = "intercepted unsigned tx"
+)
+
+type receivedTxEvent struct {
+	from                  string
+	originator            string
+	firstTimeReceivedMile int64
+	numReceived           int
+	txType                string
+	// TODO check what fields to add
+}
+
+type interceptorTxDebug struct {
+	mutex                sync.RWMutex
+	receivedTxsBroadcast map[string]*receivedTxEvent
+}
+
+func NewInterceptorTxDebug() *interceptorTxDebug {
+	return &interceptorTxDebug{
+		receivedTxsBroadcast: make(map[string]*receivedTxEvent),
+	}
+}
+
+func (id *interceptorTxDebug) Process(data process.InterceptedData, msg p2p.MessageP2P, fromConnectedPeer core.PeerID) {
+	if msg.BroadcastMethod() != p2p2.Broadcast {
+		return
+	}
+
+	isCorrectType := data.Type() == interceptedTx // && data.Type() == interceptedRewardTx && data.Type() == interceptedUnsignedTx
```

Comment:
uncomment code and use || instead of &&?

Reply:
for the moment I will let it as it is because we want to extract information  only about interceptedTx ( normal transactions)